src/dowel/csv_output.py

```diff
@@ -29,19 +30,26 @@ def record(self, data, prefix=''):
         """Log tabular data to CSV."""
         if isinstance(data, TabularInput):
             to_csv = data.as_primitive_dict
-
+            
             if not to_csv.keys() and not self._writer:
                 return
-
             if not self._writer:
-                self._fieldnames = set(to_csv.keys())
+                self._fieldnames = list(to_csv.keys())
                 self._writer = csv.DictWriter(
-                    self._log_file,
+                    self._temp_file,
                     fieldnames=self._fieldnames,
                     extrasaction='ignore')
                 self._writer.writeheader()
 
             if to_csv.keys() != self._fieldnames:
+                for key in to_csv.keys():
+                    if key not in self._fieldnames:
+                        self._fieldnames.append(key)
+                self._writer = csv.DictWriter(
+                    self._temp_file,
+                    fieldnames=self._fieldnames,
+                    extrasaction='ignore')
+
                 self._warn('Inconsistent TabularInput keys detected. '
```

Comment:
We no longer need this warning, now that we can handle the addition of new keys.